### PR TITLE
feat: added renderer for badges

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

- This PR introduces a new connector for the Badge component and associated types 
- This PR refactors the Banner component and types
- This PR introduces a new connector for the Banner component and inserts its usage on the WithBanner connector

### CODE REVIEW

File: types/features.ts [hunk 1]
```diff
@@ -5,10 +5,20 @@ export interface RSSFeed {
   blogCategory?: string;
 }
 
-export interface WebsiteBanner {
+export interface WithRange {
```
Comment (MattIPv4) [inline types/features.ts]: Does this need to be exported still?

```suggestion
interface WithRange {
```

Comment (ovflowd) [reply types/features.ts]: The WithRange? hmmm not really

Comment (MattIPv4) [reply types/features.ts]: Yeah, I figured it probably wouldn't be consumed outside inheritance within this file?

Comment (ovflowd) [reply types/features.ts]: Updated :)

**Review by AugustinMauroy — APPROVED** [review]
LGTM !

**Review by canerakdas — APPROVED** [review]
LGTM

**Pull request merged**